src/anyio/_backends/_asyncio.py

```diff
@@ -791,6 +788,7 @@ async def run_sync_in_worker_thread(
             try:
                 return await future
             finally:
+                assert worker.is_alive()
```

Comment:
Is this assert intentional?

Reply:
Yes. I found that the value was `False` in the circumstances where the hang occurred.